content/en/docs/concepts/workloads/pods/pod-lifecycle.md

```diff
@@ -460,21 +460,180 @@ An example flow:
       The containers in the Pod receive the TERM signal at different times and in an arbitrary
       order. If the order of shutdowns matters, consider using a `preStop` hook to synchronize.
       {{< /note >}}
-1. At the same time as the kubelet is starting graceful shutdown, the control plane removes that
-   shutting-down Pod from EndpointSlice (and Endpoints) objects where these represent
+1. At the same time as the kubelet is starting graceful shutdown, the control plane evaluates
+   whether a terminating Pod needs to be removed from any Endpoints of the
+   EndpointSlice object which represent
    a {{< glossary_tooltip term_id="service" text="Service" >}} with a configured
    {{< glossary_tooltip text="selector" term_id="selector" >}}.
    {{< glossary_tooltip text="ReplicaSets" term_id="replica-set" >}} and other workload resources
    no longer treat the shutting-down Pod as a valid, in-service replica. Pods that shut down slowly
-   cannot continue to serve traffic as load balancers (like the service proxy) remove the Pod from
-   the list of endpoints as soon as the termination grace period _begins_.
+   should not continue to serve regular traffic and should be terminating connection.
+   Some pods require the graceful termination - for example session draining and completion. Without the feature
+   gate `EndpointSliceTerminatingCondition` (in beta starting Kubernetes version `1.22`),
+   load balancers (like the service proxy) remove the Pod from
+   the list of Endpoints in EndpointSlices as soon as the termination grace period _begins_. With the feature
+   gate `EndpointSliceTerminatingCondition`, Endpoints are not removed from EndpointSlices for the terminating pods
+   and a status indicating [terminating state](/docs/concepts/services-networking/endpoint-slices/#conditions)
+   is exposed from Endpoint API. Terminating Endpoints always have `ready` status
+   as `false` for backward compatibility, so load balancers will not use it for regular traffic.
+   If traffic draining on terminating pod is needed, the actual readiness can be checked as a condition `serving`.
 1. When the grace period expires, the kubelet triggers forcible shutdown. The container runtime sends
    `SIGKILL` to any processes still running in any container in the Pod.
    The kubelet also cleans up a hidden `pause` container if that container runtime uses one.
 1. The kubelet triggers forcible removal of Pod object from the API server, by setting grace period
    to 0 (immediate deletion).
 1. The API server deletes the Pod's API object, which is then no longer visible from any client.
 
+
+#### Example flow with endpoint termination
+
+The following is the example of the flow described in previous section.
+
+Let's say you have a Deployment containing of a single `nginx` replica
+(just for demonstration purposes) and a Service:
+
+``` yaml
```

Comment:
looks like we don't need the extra space before 'yaml'.
Similar issues are found below.

Reply:
Could we move this into a new tutorial that follows on from [Connecting Applications With Services](https://kubernetes.io/docs/tutorials/services/connect-applications-service/)?